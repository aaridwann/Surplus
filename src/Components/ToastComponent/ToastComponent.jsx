import { get } from 'lodash';
import React, { useEffect } from 'react';
import Toast from 'react-native-toast-message';
import { useDispatch, useSelector } from 'react-redux';
import { RESET_NOTIFICATION } from '../../Redux/Features/Notification/NotificationFeature';

function ToastComponent() {
	const { notification } = useSelector((state) => state);
	const dispatch = useDispatch();

	const show = get(notification, 'isShow', false);
	const message = get(notification, 'message', '');
	const description = get(notification, 'description', '');
	const type = get(notification, 'type', 'info');

	const DetectNotification = () => {
		if (show) return showToast(message, type, description);
	};

	const showToast = (message, type, description) =>
		Toast.show({
			type: type.toLowerCase(),
			text1: message,
			text2: description,
		});

	useEffect(() => {
		DetectNotification();

		return () => dispatch(RESET_NOTIFICATION());
	}, [show]);

	return <Toast />;
}

export default ToastComponent;
