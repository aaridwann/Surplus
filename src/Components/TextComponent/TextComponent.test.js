import React from 'react';
import { render, cleanup } from '@testing-library/react-native';
import { useSelector } from 'react-redux';
import ToastComponent from './TextComponent';

// Mock useSelector
jest.mock('react-redux', () => ({
	useSelector: jest.fn(),
	useDispatch: jest.fn(),
}));

// Mock react-native-toast-message
jest.mock('react-native-toast-message', () => ({
	show: jest.fn(),
}));

// Snapshot test
describe('ToastComponent Snapshot', () => {
	afterEach(() => {
		cleanup();
		jest.clearAllMocks();
	});

	it('should match the snapshot', () => {
		useSelector.mockReturnValue({
			notification: {
				isShow: true,
				message: 'Test Message',
				description: 'Test Description',
				type: 'info',
			},
		});

		const { UNSAFE_root } = render(<ToastComponent />);
		expect(UNSAFE_root).toMatchSnapshot();
	});
});
