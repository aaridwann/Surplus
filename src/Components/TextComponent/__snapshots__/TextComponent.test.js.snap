// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToastComponent Snapshot should match the snapshot 1`] = `
ReactTestInstance {
  "_fiber": FiberNode {
    "_debugHookTypes": null,
    "_debugNeedsRemount": false,
    "_debugOwner": null,
    "_debugSource": undefined,
    "actualDuration": 0,
    "actualStartTime": -1,
    "alternate": null,
    "child": FiberNode {
      "_debugHookTypes": null,
      "_debugNeedsRemount": false,
      "_debugOwner": [Circular],
      "_debugSource": undefined,
      "actualDuration": 0,
      "actualStartTime": -1,
      "alternate": null,
      "child": FiberNode {
        "_debugHookTypes": null,
        "_debugNeedsRemount": false,
        "_debugOwner": [Circular],
        "_debugSource": null,
        "actualDuration": 0,
        "actualStartTime": -1,
        "alternate": null,
        "child": null,
        "childLanes": 0,
        "deletions": null,
        "dependencies": null,
        "elementType": "Text",
        "flags": 0,
        "index": 0,
        "key": null,
        "lanes": 0,
        "memoizedProps": Object {
          "children": undefined,
          "style": Array [
            Object {
              "fontSize": 16,
            },
            Object {},
            undefined,
          ],
        },
        "memoizedState": null,
        "mode": 0,
        "pendingProps": Object {
          "children": undefined,
          "style": Array [
            Object {
              "fontSize": 16,
            },
            Object {},
            undefined,
          ],
        },
        "ref": null,
        "return": [Circular],
        "selfBaseDuration": 0,
        "sibling": null,
        "stateNode": Object {
          "children": Array [],
          "internalInstanceHandle": [Circular],
          "isHidden": false,
          "props": Object {
            "children": undefined,
            "style": Array [
              Object {
                "fontSize": 16,
              },
              Object {},
              undefined,
            ],
          },
          "rootContainerInstance": Object {
            "children": Array [
              [Circular],
            ],
            "createNodeMock": [Function],
            "tag": "CONTAINER",
          },
          "tag": "INSTANCE",
          "type": "Text",
        },
        "subtreeFlags": 0,
        "tag": 5,
        "treeBaseDuration": 0,
        "type": "Text",
        "updateQueue": null,
      },
      "childLanes": 0,
      "deletions": null,
      "dependencies": null,
      "elementType": [Function],
      "flags": 1,
      "index": 0,
      "key": null,
      "lanes": 0,
      "memoizedProps": Object {
        "style": Array [
          Object {
            "fontSize": 16,
          },
          Object {},
          undefined,
        ],
      },
      "memoizedState": null,
      "mode": 0,
      "pendingProps": Object {
        "style": Array [
          Object {
            "fontSize": 16,
          },
          Object {},
          undefined,
        ],
      },
      "ref": null,
      "return": [Circular],
      "selfBaseDuration": 0,
      "sibling": null,
      "stateNode": Component {
        "_reactInternalInstance": Object {},
        "_reactInternals": [Circular],
        "context": Object {},
        "props": Object {
          "style": Array [
            Object {
              "fontSize": 16,
            },
            Object {},
            undefined,
          ],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
        },
      },
      "subtreeFlags": 0,
      "tag": 1,
      "treeBaseDuration": 0,
      "type": [Function],
      "updateQueue": Object {
        "baseState": null,
        "effects": null,
        "firstBaseUpdate": null,
        "lastBaseUpdate": null,
        "shared": Object {
          "interleaved": null,
          "lanes": 0,
          "pending": null,
        },
      },
    },
    "childLanes": 0,
    "deletions": null,
    "dependencies": null,
    "elementType": [Function],
    "flags": 1,
    "index": 0,
    "key": null,
    "lanes": 0,
    "memoizedProps": Object {},
    "memoizedState": null,
    "mode": 0,
    "pendingProps": Object {},
    "ref": null,
    "return": FiberNode {
      "_debugHookTypes": null,
      "_debugNeedsRemount": false,
      "_debugOwner": null,
      "_debugSource": null,
      "actualDuration": 0,
      "actualStartTime": -1,
      "alternate": FiberNode {
        "_debugHookTypes": null,
        "_debugNeedsRemount": false,
        "_debugOwner": null,
        "_debugSource": null,
        "actualDuration": 0,
        "actualStartTime": -1,
        "alternate": [Circular],
        "child": null,
        "childLanes": 0,
        "deletions": null,
        "dependencies": null,
        "elementType": null,
        "flags": 0,
        "index": 0,
        "key": null,
        "lanes": 1,
        "memoizedProps": null,
        "memoizedState": Object {
          "cache": null,
          "element": null,
          "isDehydrated": false,
          "pendingSuspenseBoundaries": null,
          "transitions": null,
        },
        "mode": 0,
        "pendingProps": null,
        "ref": null,
        "return": null,
        "selfBaseDuration": 0,
        "sibling": null,
        "stateNode": FiberRootNode {
          "_debugRootType": "render()",
          "callbackNode": null,
          "callbackPriority": 0,
          "containerInfo": Object {
            "children": Array [
              Object {
                "children": Array [],
                "internalInstanceHandle": FiberNode {
                  "_debugHookTypes": null,
                  "_debugNeedsRemount": false,
                  "_debugOwner": FiberNode {
                    "_debugHookTypes": null,
                    "_debugNeedsRemount": false,
                    "_debugOwner": [Circular],
                    "_debugSource": undefined,
                    "actualDuration": 0,
                    "actualStartTime": -1,
                    "alternate": null,
                    "child": [Circular],
                    "childLanes": 0,
                    "deletions": null,
                    "dependencies": null,
                    "elementType": [Function],
                    "flags": 1,
                    "index": 0,
                    "key": null,
                    "lanes": 0,
                    "memoizedProps": Object {
                      "style": Array [
                        Object {
                          "fontSize": 16,
                        },
                        Object {},
                        undefined,
                      ],
                    },
                    "memoizedState": null,
                    "mode": 0,
                    "pendingProps": Object {
                      "style": Array [
                        Object {
                          "fontSize": 16,
                        },
                        Object {},
                        undefined,
                      ],
                    },
                    "ref": null,
                    "return": [Circular],
                    "selfBaseDuration": 0,
                    "sibling": null,
                    "stateNode": Component {
                      "_reactInternalInstance": Object {},
                      "_reactInternals": [Circular],
                      "context": Object {},
                      "props": Object {
                        "style": Array [
                          Object {
                            "fontSize": 16,
                          },
                          Object {},
                          undefined,
                        ],
                      },
                      "refs": Object {},
                      "state": null,
                      "updater": Object {
                        "enqueueForceUpdate": [Function],
                        "enqueueReplaceState": [Function],
                        "enqueueSetState": [Function],
                        "isMounted": [Function],
                      },
                    },
                    "subtreeFlags": 0,
                    "tag": 1,
                    "treeBaseDuration": 0,
                    "type": [Function],
                    "updateQueue": Object {
                      "baseState": null,
                      "effects": null,
                      "firstBaseUpdate": null,
                      "lastBaseUpdate": null,
                      "shared": Object {
                        "interleaved": null,
                        "lanes": 0,
                        "pending": null,
                      },
                    },
                  },
                  "_debugSource": null,
                  "actualDuration": 0,
                  "actualStartTime": -1,
                  "alternate": null,
                  "child": null,
                  "childLanes": 0,
                  "deletions": null,
                  "dependencies": null,
                  "elementType": "Text",
                  "flags": 0,
                  "index": 0,
                  "key": null,
                  "lanes": 0,
                  "memoizedProps": Object {
                    "children": undefined,
                    "style": Array [
                      Object {
                        "fontSize": 16,
                      },
                      Object {},
                      undefined,
                    ],
                  },
                  "memoizedState": null,
                  "mode": 0,
                  "pendingProps": Object {
                    "children": undefined,
                    "style": Array [
                      Object {
                        "fontSize": 16,
                      },
                      Object {},
                      undefined,
                    ],
                  },
                  "ref": null,
                  "return": FiberNode {
                    "_debugHookTypes": null,
                    "_debugNeedsRemount": false,
                    "_debugOwner": [Circular],
                    "_debugSource": undefined,
                    "actualDuration": 0,
                    "actualStartTime": -1,
                    "alternate": null,
                    "child": [Circular],
                    "childLanes": 0,
                    "deletions": null,
                    "dependencies": null,
                    "elementType": [Function],
                    "flags": 1,
                    "index": 0,
                    "key": null,
                    "lanes": 0,
                    "memoizedProps": Object {
                      "style": Array [
                        Object {
                          "fontSize": 16,
                        },
                        Object {},
                        undefined,
                      ],
                    },
                    "memoizedState": null,
                    "mode": 0,
                    "pendingProps": Object {
                      "style": Array [
                        Object {
                          "fontSize": 16,
                        },
                        Object {},
                        undefined,
                      ],
                    },
                    "ref": null,
                    "return": [Circular],
                    "selfBaseDuration": 0,
                    "sibling": null,
                    "stateNode": Component {
                      "_reactInternalInstance": Object {},
                      "_reactInternals": [Circular],
                      "context": Object {},
                      "props": Object {
                        "style": Array [
                          Object {
                            "fontSize": 16,
                          },
                          Object {},
                          undefined,
                        ],
                      },
                      "refs": Object {},
                      "state": null,
                      "updater": Object {
                        "enqueueForceUpdate": [Function],
                        "enqueueReplaceState": [Function],
                        "enqueueSetState": [Function],
                        "isMounted": [Function],
                      },
                    },
                    "subtreeFlags": 0,
                    "tag": 1,
                    "treeBaseDuration": 0,
                    "type": [Function],
                    "updateQueue": Object {
                      "baseState": null,
                      "effects": null,
                      "firstBaseUpdate": null,
                      "lastBaseUpdate": null,
                      "shared": Object {
                        "interleaved": null,
                        "lanes": 0,
                        "pending": null,
                      },
                    },
                  },
                  "selfBaseDuration": 0,
                  "sibling": null,
                  "stateNode": [Circular],
                  "subtreeFlags": 0,
                  "tag": 5,
                  "treeBaseDuration": 0,
                  "type": "Text",
                  "updateQueue": null,
                },
                "isHidden": false,
                "props": Object {
                  "children": undefined,
                  "style": Array [
                    Object {
                      "fontSize": 16,
                    },
                    Object {},
                    undefined,
                  ],
                },
                "rootContainerInstance": [Circular],
                "tag": "INSTANCE",
                "type": "Text",
              },
            ],
            "createNodeMock": [Function],
            "tag": "CONTAINER",
          },
          "context": Object {},
          "current": [Circular],
          "effectDuration": 0,
          "entangledLanes": 0,
          "entanglements": Array [
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
          ],
          "eventTimes": Array [
            -1,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
          ],
          "expirationTimes": Array [
            -1,
            -1,
            -1,
            -1,
            -1,
            -1,
            -1,
            -1,
            -1,
            -1,
            -1,
            -1,
            -1,
            -1,
            -1,
            -1,
            -1,
            -1,
            -1,
            -1,
            -1,
            -1,
            -1,
            -1,
            -1,
            -1,
            -1,
            -1,
            -1,
            -1,
            -1,
          ],
          "expiredLanes": 0,
          "finishedLanes": 0,
          "finishedWork": null,
          "identifierPrefix": "",
          "mutableReadLanes": 0,
          "onRecoverableError": [Function],
          "passiveEffectDuration": 0,
          "pendingChildren": null,
          "pendingContext": null,
          "pendingLanes": 0,
          "pingCache": null,
          "pingedLanes": 0,
          "suspendedLanes": 0,
          "tag": 0,
          "timeoutHandle": -1,
        },
        "subtreeFlags": 0,
        "tag": 3,
        "treeBaseDuration": 0,
        "type": null,
        "updateQueue": Object {
          "baseState": Object {
            "cache": null,
            "element": null,
            "isDehydrated": false,
            "pendingSuspenseBoundaries": null,
            "transitions": null,
          },
          "effects": null,
          "firstBaseUpdate": Object {
            "callback": null,
            "eventTime": 1685641938880,
            "lane": 1,
            "next": null,
            "payload": Object {
              "element": <CustomText />,
            },
            "tag": 0,
          },
          "lastBaseUpdate": Object {
            "callback": null,
            "eventTime": 1685641938880,
            "lane": 1,
            "next": null,
            "payload": Object {
              "element": <CustomText />,
            },
            "tag": 0,
          },
          "shared": Object {
            "interleaved": null,
            "lanes": 0,
            "pending": null,
          },
        },
      },
      "child": [Circular],
      "childLanes": 0,
      "deletions": null,
      "dependencies": null,
      "elementType": null,
      "flags": 1024,
      "index": 0,
      "key": null,
      "lanes": 0,
      "memoizedProps": null,
      "memoizedState": Object {
        "cache": null,
        "element": <CustomText />,
        "isDehydrated": false,
        "pendingSuspenseBoundaries": null,
        "transitions": null,
      },
      "mode": 0,
      "pendingProps": null,
      "ref": null,
      "return": null,
      "selfBaseDuration": 0,
      "sibling": null,
      "stateNode": FiberRootNode {
        "_debugRootType": "render()",
        "callbackNode": null,
        "callbackPriority": 0,
        "containerInfo": Object {
          "children": Array [
            Object {
              "children": Array [],
              "internalInstanceHandle": FiberNode {
                "_debugHookTypes": null,
                "_debugNeedsRemount": false,
                "_debugOwner": FiberNode {
                  "_debugHookTypes": null,
                  "_debugNeedsRemount": false,
                  "_debugOwner": [Circular],
                  "_debugSource": undefined,
                  "actualDuration": 0,
                  "actualStartTime": -1,
                  "alternate": null,
                  "child": [Circular],
                  "childLanes": 0,
                  "deletions": null,
                  "dependencies": null,
                  "elementType": [Function],
                  "flags": 1,
                  "index": 0,
                  "key": null,
                  "lanes": 0,
                  "memoizedProps": Object {
                    "style": Array [
                      Object {
                        "fontSize": 16,
                      },
                      Object {},
                      undefined,
                    ],
                  },
                  "memoizedState": null,
                  "mode": 0,
                  "pendingProps": Object {
                    "style": Array [
                      Object {
                        "fontSize": 16,
                      },
                      Object {},
                      undefined,
                    ],
                  },
                  "ref": null,
                  "return": [Circular],
                  "selfBaseDuration": 0,
                  "sibling": null,
                  "stateNode": Component {
                    "_reactInternalInstance": Object {},
                    "_reactInternals": [Circular],
                    "context": Object {},
                    "props": Object {
                      "style": Array [
                        Object {
                          "fontSize": 16,
                        },
                        Object {},
                        undefined,
                      ],
                    },
                    "refs": Object {},
                    "state": null,
                    "updater": Object {
                      "enqueueForceUpdate": [Function],
                      "enqueueReplaceState": [Function],
                      "enqueueSetState": [Function],
                      "isMounted": [Function],
                    },
                  },
                  "subtreeFlags": 0,
                  "tag": 1,
                  "treeBaseDuration": 0,
                  "type": [Function],
                  "updateQueue": Object {
                    "baseState": null,
                    "effects": null,
                    "firstBaseUpdate": null,
                    "lastBaseUpdate": null,
                    "shared": Object {
                      "interleaved": null,
                      "lanes": 0,
                      "pending": null,
                    },
                  },
                },
                "_debugSource": null,
                "actualDuration": 0,
                "actualStartTime": -1,
                "alternate": null,
                "child": null,
                "childLanes": 0,
                "deletions": null,
                "dependencies": null,
                "elementType": "Text",
                "flags": 0,
                "index": 0,
                "key": null,
                "lanes": 0,
                "memoizedProps": Object {
                  "children": undefined,
                  "style": Array [
                    Object {
                      "fontSize": 16,
                    },
                    Object {},
                    undefined,
                  ],
                },
                "memoizedState": null,
                "mode": 0,
                "pendingProps": Object {
                  "children": undefined,
                  "style": Array [
                    Object {
                      "fontSize": 16,
                    },
                    Object {},
                    undefined,
                  ],
                },
                "ref": null,
                "return": FiberNode {
                  "_debugHookTypes": null,
                  "_debugNeedsRemount": false,
                  "_debugOwner": [Circular],
                  "_debugSource": undefined,
                  "actualDuration": 0,
                  "actualStartTime": -1,
                  "alternate": null,
                  "child": [Circular],
                  "childLanes": 0,
                  "deletions": null,
                  "dependencies": null,
                  "elementType": [Function],
                  "flags": 1,
                  "index": 0,
                  "key": null,
                  "lanes": 0,
                  "memoizedProps": Object {
                    "style": Array [
                      Object {
                        "fontSize": 16,
                      },
                      Object {},
                      undefined,
                    ],
                  },
                  "memoizedState": null,
                  "mode": 0,
                  "pendingProps": Object {
                    "style": Array [
                      Object {
                        "fontSize": 16,
                      },
                      Object {},
                      undefined,
                    ],
                  },
                  "ref": null,
                  "return": [Circular],
                  "selfBaseDuration": 0,
                  "sibling": null,
                  "stateNode": Component {
                    "_reactInternalInstance": Object {},
                    "_reactInternals": [Circular],
                    "context": Object {},
                    "props": Object {
                      "style": Array [
                        Object {
                          "fontSize": 16,
                        },
                        Object {},
                        undefined,
                      ],
                    },
                    "refs": Object {},
                    "state": null,
                    "updater": Object {
                      "enqueueForceUpdate": [Function],
                      "enqueueReplaceState": [Function],
                      "enqueueSetState": [Function],
                      "isMounted": [Function],
                    },
                  },
                  "subtreeFlags": 0,
                  "tag": 1,
                  "treeBaseDuration": 0,
                  "type": [Function],
                  "updateQueue": Object {
                    "baseState": null,
                    "effects": null,
                    "firstBaseUpdate": null,
                    "lastBaseUpdate": null,
                    "shared": Object {
                      "interleaved": null,
                      "lanes": 0,
                      "pending": null,
                    },
                  },
                },
                "selfBaseDuration": 0,
                "sibling": null,
                "stateNode": [Circular],
                "subtreeFlags": 0,
                "tag": 5,
                "treeBaseDuration": 0,
                "type": "Text",
                "updateQueue": null,
              },
              "isHidden": false,
              "props": Object {
                "children": undefined,
                "style": Array [
                  Object {
                    "fontSize": 16,
                  },
                  Object {},
                  undefined,
                ],
              },
              "rootContainerInstance": [Circular],
              "tag": "INSTANCE",
              "type": "Text",
            },
          ],
          "createNodeMock": [Function],
          "tag": "CONTAINER",
        },
        "context": Object {},
        "current": [Circular],
        "effectDuration": 0,
        "entangledLanes": 0,
        "entanglements": Array [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
        "eventTimes": Array [
          -1,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
        "expirationTimes": Array [
          -1,
          -1,
          -1,
          -1,
          -1,
          -1,
          -1,
          -1,
          -1,
          -1,
          -1,
          -1,
          -1,
          -1,
          -1,
          -1,
          -1,
          -1,
          -1,
          -1,
          -1,
          -1,
          -1,
          -1,
          -1,
          -1,
          -1,
          -1,
          -1,
          -1,
          -1,
        ],
        "expiredLanes": 0,
        "finishedLanes": 0,
        "finishedWork": null,
        "identifierPrefix": "",
        "mutableReadLanes": 0,
        "onRecoverableError": [Function],
        "passiveEffectDuration": 0,
        "pendingChildren": null,
        "pendingContext": null,
        "pendingLanes": 0,
        "pingCache": null,
        "pingedLanes": 0,
        "suspendedLanes": 0,
        "tag": 0,
        "timeoutHandle": -1,
      },
      "subtreeFlags": 3,
      "tag": 3,
      "treeBaseDuration": 0,
      "type": null,
      "updateQueue": Object {
        "baseState": Object {
          "cache": null,
          "element": <CustomText />,
          "isDehydrated": false,
          "pendingSuspenseBoundaries": null,
          "transitions": null,
        },
        "effects": null,
        "firstBaseUpdate": null,
        "lastBaseUpdate": null,
        "shared": Object {
          "interleaved": null,
          "lanes": 0,
          "pending": null,
        },
      },
    },
    "selfBaseDuration": 0,
    "sibling": null,
    "stateNode": null,
    "subtreeFlags": 1,
    "tag": 0,
    "treeBaseDuration": 0,
    "type": [Function],
    "updateQueue": null,
  },
}
`;
